e2etest/declarativeRunner.go

```diff
@@ -54,34 +118,43 @@ func RunScenarios(
 	}
 
 	// construct all the scenarios
-	scenarios := make([]scenario, 0, 16)
+	scenarios := make([][]scenario, 0, 16)
 	for _, op := range operations.getValues() {
 		for _, fromTo := range testFromTo.getValues(op) {
-			// Create unique name for generating container names
-			compactScenarioName := fmt.Sprintf("%.4s-%s-%c-%c%c", suiteName, testName, op.String()[0], fromTo.From().String()[0], fromTo.To().String()[0])
-			fullScenarioName := fmt.Sprintf("%s.%s.%s-%s", suiteName, testName, op.String(), fromTo.String())
-			// Sub-test name is not globally unique (it doesn't need to be) but it is more human-readable
-			subtestName := fmt.Sprintf("%s-%s", op, fromTo)
-
-			hsToUse := hooks{}
-			if hs != nil {
-				hsToUse = *hs
+			credentialTypes := getValidCredCombinationsForFromTo(fromTo, requestedCredentialTypesSrc, requestedCredentialTypesDst)
+
+			scenarioList := make([]scenario, 0)
+
+			for _, credTypes := range credentialTypes {
```

Comment:
It might be an overkill to execute both credential types for all testcases. Can we run oauth on selective scenarios? Like the tests which focus on Frontend filters - execute same path on both credentials